src/workerd/api/actor-state.c++

```diff
@@ -492,11 +493,19 @@ jsg::Promise<void> DurableObjectStorage::deleteAll(
 
   auto deleteAll = cache->deleteAll(options);
 
-  auto& context = IoContext::current();
-  context.addTask(deleteAll.count.then([&metrics = currentActorMetrics()](uint deleted) {
+  const auto updateStorageDeletes = [&](IoContext& context,
```

Comment:
Did you mean to capture `[]` rather than `[&]`? We should only be capturing `[&]` in lambdas which run entirely synchronously.

Reply:
ah right. Muscle memory is fun ;-) ... I've converted this to not use a capturing lambda to highlight that the `IoContext::runTask(...)` is intended to work with more than just lambdas. That said, as mentioned above, I don't feel strongly about adding this and I'm happy to back this particular change back out.